### ISSUE TITLE
Using aria-describedby to provide a long description of an img does not work (for my tests); why or should I not try and just use longdesc or aria-labelledby?

### ISSUE THREAD

**Issue by alanfluff** [nvaccess/nvda#9534]
I asked this [over here (5398)](https://github.com/nvaccess/nvda/issues/5398
), but it is a closed issue which means it won't be seen (?is that right?) - so referencing it here in a new issue.

I may be totally wrong and asking a Q on a closed issue is as valid as asking in a new open issue, in which case, apologies to mods and please feel free to delete this issue.

Thanks.

**Comment by leonardder** [comment]
I could be wrong, but it looks like the issue description is missing your actual question.

**Comment by alanfluff** [comment]
Thanks @leonardder (duh, what am I like)

**Comment by ehollig** [comment]
@alanfluff could you please update your original issue filling out the complete issue template so we may understand the issue you are experiencing? Alternatively, if this is a generic question, you can ask on the [NVDA Screen Reader Development nvda-devel@groups.io mailing list](https://groups.io/g/nvda-devel).

**Comment by Adriani90** [comment]
@alanfluff did you ask this on the developers mailing list? Alternatively you could also ask your question here:
https://gitter.im/nvaccess/NVDA

I am closing this issue for now.

**Comment by alanfluff** [comment]
@Adriani90 Sorry for the radio silence. I am embarrassed to say I can't remember where I posted it.

However, since posting this I found a couple of items that suggested this attribute is not simply read out in all circumstances, notably [Gotchas when describing content](https://developer.paciellogroup.com/blog/2018/09/describing-aria-describedby/) and *at one time* (before an edit), a similar story [here](https://www.w3.org/WAI/tutorials/forms/instructions/); however this second page has since been edited and no longer has the text I noted (and [this question](https://stackoverflow.com/questions/56465819/accessibility-in-what-scenarios-would-aria-describedby-not-be-announced) also noted).

So as far as I am concerned, please leave this closed.

My, slightly unsatisfactory take-away (with the technology in general, not NVDA), is that `aria-labelledby` is reliable and so I used it, and `aria-described-by` is quirky so I ignore it.

**Comment by Adriani90** [comment]
Thank you very much for your reporting Alan. Very appreciated. This is well noted.

 

Von: Alan <notifications@github.com> 
Gesendet: Dienstag, 28. Januar 2020 20:16
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Mention <mention@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Using aria-describedby to provide a long description of an img does not work (for my tests); why or should I not try and just use longdesc or aria-labelledby? (#9534)

 

@Adriani90 <https://github.com/Adriani90>  Sorry for the radio silence. I am embarrassed to say I can't remember where I posted it.

However, since posting this I found a couple of items that suggested this attribute is not simply read out in all circumstances, notably Gotchas when describing content <https://developer.paciellogroup.com/blog/2018/09/describing-aria-describedby/>  and at one time (before an edit), a similar story here <https://www.w3.org/WAI/tutorials/forms/instructions/> ; however this second page has since been edited and no longer has the text I noted (and this question <https://stackoverflow.com/questions/56465819/accessibility-in-what-scenarios-would-aria-describedby-not-be-announced>  also noted).

So as far as I am concerned, please leave this closed.

My, slightly unsatisfactory take-away (with the technology in general, not NVDA), is that aria-labelledby is reliable and so I used it, and aria-described-by is quirky so I ignore it.
